commit 6181b0c6432bf0807512e85e0c5863f7aca8e515
Author: Chuck Lever <chuck.lever@oracle.com>
Date:   Mon Jan 2 12:05:56 2023 -0500

    SUNRPC: Convert svcauth_unix_accept() to use xdr_stream
    
    Done as part of hardening the server-side RPC header decoding path.
    
    Since the server-side of the Linux kernel SunRPC implementation
    ignores the contents of the Call's machinename field, there's no
    need for its RPC_AUTH_UNIX authenticator to reject names that are
    larger than UNX_MAXNODENAME.
    
    Reviewed-by: Jeff Layton <jlayton@kernel.org>
    Signed-off-by: Chuck Lever <chuck.lever@oracle.com>

diff --git a/include/linux/sunrpc/msg_prot.h b/include/linux/sunrpc/msg_prot.h
index 02117ed0fa2e..c4b0eb2b2f04 100644
--- a/include/linux/sunrpc/msg_prot.h
+++ b/include/linux/sunrpc/msg_prot.h
@@ -34,6 +34,11 @@ enum rpc_auth_flavors {
 	RPC_AUTH_GSS_SPKMP = 390011,
 };
 
+/* Maximum size (in octets) of the machinename in an AUTH_UNIX
+ * credential (per RFC 5531 Appendix A)
+ */
+#define RPC_MAX_MACHINENAME	(255)
+
 /* Maximum size (in bytes) of an rpc credential or verifier */
 #define RPC_MAX_AUTH_SIZE (400)